File: src/layer_04_infrastructure/ui/web/frontend/hooks/use_generate_document_from_template.js
import { GenerateDocumentFromTemplateController } from "../../../layer_03_interface_adapters/controllers/web/generate_document_from_template.js";

/**
 * Custom Hook quản lý trạng thái hiển thị và luồng công việc cho tính năng GenerateDocumentFromTemplate.
 */
export class UseGenerateDocumentFromTemplate extends EventTarget {
    constructor() {
        super();
        this.loading = false;
        this.data = {};
    }
}

File: src/layer_04_infrastructure/ui/desktop_qt6/hooks/use_generate_document_from_template.py
from PyQt6.QtCore import QObject, pyqtSignal
from src.layer_03_interface_adapters.controllers.desktop.generate_document_from_template import (
    GenerateDocumentFromTemplateController,
)


class UseGenerateDocumentFromTemplate(QObject):
    """
    Custom Hook quản lý trạng thái hiển thị và luồng công việc cho tính năng GenerateDocumentFromTemplate.
    """

    data_changed = pyqtSignal(dict)
    loading = pyqtSignal(bool)
    error = pyqtSignal(str)

    def __init__(self, parent=None):
        super().__init__(parent)
